### ISSUE TITLE
Contact

### ISSUE THREAD

**Issue by pre-vail** [vekonyz/fast-protocol#4]
How can i contact you? I'd like to learn a little more about FAST and possibly work on this project. It would be nice if we can have a short phone conversation.

**Comment by vekonyz** [comment]
You find the specification here https://www.fixtrading.org/standards/fast/

**Comment by pre-vail** [comment]
Thanks I’m aware of that page. Can you give a brief description of how you would go about implementing support for streaming?

**Comment by vekonyz** [comment]
streaming with or without additional framing?

**Comment by pre-vail** [comment]
What do you think is the best option?

**Comment by vekonyz** [comment]
What is your use case?

**Comment by pre-vail** [comment]
https://gainfutures.com/wp-content/uploads/FIX.pdf
Pages 49 to the end explain what I’m using FAST for.

**Comment by pre-vail** [comment]
Have you had a chance to take a look at the pdf?

**Comment by vekonyz** [comment]
Yes, I didn't found any references in regards of additional framing, but found an issue: in MarketDataSnapshotFullRefresh message the field MDEntryPx uses exponent/mantissa operator for encoding decimal values - and this is still not implemented here ...

**Comment by pre-vail** [comment]
How should it be implemented? Can you give some insight? Maybe it will be the first thing I work on.

**Comment by vekonyz** [comment]
The idea in this FAST implementation is that after loading the xml specification the message structures are stored in memory and each time a message is encoded or decoded, the program iterates over the stored data 'model'. In order to support handling exponent and mantissa individually, you have to add special processing rules in encoder as well as in decoder part. The problem with individual matissa and exponent is that each of them can have potentially individual operators and constraints...

**Comment by pre-vail** [comment]
Thank you for that explanation. A little background on my knowledge of FAST up to this point: I first heard of both FIX and FAST about 2-3 months ago so I’m very new to both of them. FIX is not that hard to understand, I view it as name/value pairs joined by a separator. I’m using the fixparser <https://www.npmjs.com/package/fixparser> package to handle FIX messages. FAST is a different story. I have a list <https://www.fixtrading.org/standards/fast> of documents that I want to read in order to understand it. In particular, the documents are:

1 FAST Specification Version 1.1			(reading now)
2 FAST Session Control Protocol (SCP) 1.1
3 Basic FAST User's Guide
4 FAST Protocol Technical Overview		(quickly reviewed it but will read it again)
5 FIX Over FAST Implementation Guide	(done)
6 FAST Extension Version 1.2			(not needed but i’ll read it anyway)

At this point, I honestly don’t understand what you mean by “each of them can have potentially individual operators and constraints” but that’s probably due to the fact that I don’t understand FAST well enough yet. I’ll continue reading from my list. If you can clarify that statement, I’d appreciate it. If you feel that I just need to learn more about FAST, I understand that too. Either way, thank you for answering my questions so far, I look forward to adding any necessary functionality to this package as I need it to continue with the project that I’m working on.

**Comment by pre-vail** [comment]
Hello. Can you clarify your statement or should I just keep learning about FAST?

**Comment by vekonyz** [comment]
Studying the FAST specification is the right way to become more insights into the protocol. You will find also the definition of the mantissa/exponent handling there.

**Comment by pre-vail** [comment]
Hello. OK, so I've read up on FAST and its Session Control Protocol. There is still a lot of understanding left for me to do. Hopefully you can help by providing an example of the file that is being read on line 5, and the template from line 7, of samples/decode.js? Can you also provide an example of the file that is being read on line 14, and the template from line 19, of samples/gw/gwlogreader.js?
